### PR TITLE
Ignore errors for `try` pushes (Fixes  #587)

### LINKED ISSUE #587: `push.classify_regressions()` produce errors originating from a parent's revision
For a given push object, for example, push = Push('b6a94bff9d8e0c31fd28b848bbad84a23731e9c0','try') ,
`push.classify_regressions()`  produced some errors. Some of the error logs are seen below.  `xpcshell.ini` files originate from the parent revision.

19:07:08.561 | ERROR    | mozci.data.sources.artifact:64  - Some groups in FR_bhpkAR7-FKdrYThH7Xw are missing results: 
{'browser/components/migration/tests/unit/xpcshell.ini', 'devtools/shared/qrcode/tests/xpcshell/xpcshell.ini'}

`19:07:09.000 | ERROR    | mozci.data.sources.artifact:64  - Some groups in ehrnCDlqQju3CeooLGutSg are missing results: {'browser/base/content/test/plugins/browser.ini', 'dom/cache/test/browser/browser.ini', 'dom/u2f/tests/browser/browser.ini'}

`19:07:28.832 | ERROR    | mozci.data.sources.artifact:64  - Some groups in OSIBkgxCRTWjK4XXGDKr1Q are missing results: {'editor/txmgr/tests/crashtests/crashtests.list', 'gfx/tests/crashtests/crashtests.list', 'editor/libeditor/crashtests/crashtests.list'}

19:07:28.996 | ERROR    | mozci.data.sources.artifact:64  - Some groups in Lb8VQ42KTseWLWKGhXaxiQ are missing results: {'dom/mathml/crashtests/crashtests.list', 'layout/generic/crashtests/crashtests.list'}

### PR DESCRIPTION
**Pull request by chichi012**
Fixes  #587 

Skipped error logs for pushes on the `try` branch.

### CODE REVIEW

**Comment by chichi012** [comment]
@marco-c  @jmaher if you have some time, kindly help review this pr

**Comment by jmaher** [comment]
overall, I have no problem with this- I will let @marco-c determine if this is the right fix or if there is something else to do.

**Comment by marco-c** [comment]
Instead of doing something specific for try, we could turn the error into a warning or a debug message. Wdyt @ahal?

**Comment by ahal** [comment]
Yeah, this error is much more verbose for try pushes since the parent is typically a mozilla-central push. Maybe we could make it a `debug` message for try and a `warning` otherwise?

**Comment by ahal** [comment]
Though if you think it could just be a debug message everywhere, that's fine with me.

File: mozci/data/sources/artifact/__init__.py [hunk 1]
```diff
@@ -61,9 +62,10 @@ def _load_errorsummary(self, task_id) -> None:
 
         missing_groups = groups - set(group_results)
         if len(missing_groups) > 0:
-            logger.error(
-                f"Some groups in {task_id} are missing results: {missing_groups}"
-            )
+            if not self.is_try:
+                logger.error(
```
Comment (ahal) [inline mozci/data/sources/artifact/__init__.py:66]: Let's turn this into a `warning` message for the not try case.

File: mozci/data/sources/artifact/__init__.py [hunk 2]
```diff
@@ -61,9 +62,10 @@ def _load_errorsummary(self, task_id) -> None:
...
+                    f"Some groups in {task_id} are missing results: {missing_groups}"
+                )
```
Comment (ahal) [inline mozci/data/sources/artifact/__init__.py:68]: We should continue to at least log this at the `debug` level for try pushes. According to [the loguru docs](https://loguru.readthedocs.io/en/stable/api/logger.html#loguru._logger.Logger.log) you should be able to do something like `logger.log(LEVEL, msg)` (where LEVEL is "DEBUG" or "WARNING").

**Review by ahal — CHANGES_REQUESTED** [review]

**Review by jmaher — APPROVED** [review]

**Pull request merged**